python/hail/methods/qc.py

```diff
@@ -169,3 +169,512 @@ def variant_qc(dataset, name='variant_qc'):
     """
 
     return MatrixTable(Env.hail().methods.VariantQC.apply(dataset._jvds, 'va.`{}`'.format(name)))
+
+
+@handle_py4j
+@typecheck(dataset=MatrixTable,
+           config=strlike,
+           block_size=integral,
+           name=strlike,
+           csq=bool)
+def vep(dataset, config, block_size=1000, name='vep', csq=False):
+    """Annotate variants with VEP.
+
+    .. include:: ../_templates/req_tvariant.rst
+
+    :py:meth:`~hail.methods.vep` runs `Variant Effect Predictor
+    <http://www.ensembl.org/info/docs/tools/vep/index.html>`__ with the `LOFTEE
+    plugin <https://github.com/konradjk/loftee>`__ on the current dataset and
+    adds the result as a row field.
+
+    Examples
+    --------
+
+    Add VEP annotations to the dataset:
+
+    >>> result = methods.vep(dataset, "data/vep.properties") # doctest: +SKIP
+
+    Notes
+    -----
+
+    **Configuration**
+
+    :py:meth:`~hail.VariantDataset.vep` needs a configuration file to tell it
+    how to run VEP. The format is a `.properties file
+    <https://en.wikipedia.org/wiki/.properties>`__. Roughly, each line defines a
+    property as a key-value pair of the form `key = value`. `vep` supports the
+    following properties:
+
+    - **hail.vep.perl** -- Location of Perl. Optional, default: perl.
+    - **hail.vep.perl5lib** -- Value for the PERL5LIB environment variable when
+      invoking VEP. Optional, by default PERL5LIB is not set.
+    - **hail.vep.path** -- Value of the PATH environment variable when invoking
+      VEP.  Optional, by default PATH is not set.
+    - **hail.vep.location** -- Location of the VEP Perl script.  Required.
+    - **hail.vep.cache_dir** -- Location of the VEP cache dir, passed to VEP
+      with the `--dir` option. Required.
+    - **hail.vep.fasta** -- Location of the FASTA file to use to look up the
+      reference sequence, passed to VEP with the `--fasta` option. Required.
+    - **hail.vep.assembly** -- Genome assembly version to use. Optional,
+      default: GRCh37
+    - **hail.vep.plugin** -- VEP plugin, passed to VEP with the `--plugin`
+      option. Optional. Overrides `hail.vep.lof.human_ancestor` and
+      `hail.vep.lof.conservation_file`.
+    - **hail.vep.lof.human_ancestor** -- Location of the human ancestor file for
+      the LOFTEE plugin. Ignored if `hail.vep.plugin` is set. Required otherwise.
+    - **hail.vep.lof.conservation_file** -- Location of the conservation file
```

Comment:
My understanding is that all of these Hail field paths should be italics, which can, apparently, be achieved with single-asterisk or with single-backtick, though the meaning of single-backtick is unclear to me.

Reply:
@danking these aren't hail field paths. They're keys in the properties file.

@tpoterba The format I have currently is what we decided upon for definition lists a year ago. We could change it to the definition list format (http://www.sphinx-doc.org/en/stable/rest.html#lists-and-quote-like-blocks).

Reply:
Ah. I guess I'm ambivalent then. Using sphinx-style definition lists seems like a Good Thing (tm).